src/schema/types/person.ts

```diff
@@ -17,21 +17,41 @@ import {
 
 import { globalIdField } from 'graphql-relay';
 
-import { nodeInterface } from '../relayNode';
-import { createdField, editedField } from '../commonFields';
-import { connectionFromUrls } from '../connections';
-import { getObjectFromUrl, convertToNumber } from '../apiHelper';
+import { nodeInterface } from '../relayNode.js';
+import { createdField, editedField } from '../commonFields.js';
+import { connectionFromUrls } from '../connections.js';
+import { getObjectFromUrl, convertToNumber } from '../apiHelper.js';
+
+import FilmType from './film.js';
+import PlanetType from './planet.js';
+import SpeciesType from './species.js';
+import StarshipType from './starship.js';
+import VehicleType from './vehicle.js';
+
 
-import FilmType from './film';
-import PlanetType from './planet';
-import SpeciesType from './species';
-import StarshipType from './starship';
-import VehicleType from './vehicle';
+type Person = {
+  name: string,
+  birth_year: string,
+  eye_color: string,
+  hair_color: string,
+  height: string,
+  mass: string,
+  skin_color: string,
+  homeworld: string,
+  films: Array<string>,
+  species: Array<string>,
+  starships: Array<string>,
+  vehicles: Array<string>,
+  url: string,
+  created: string,
+  edited: string,
+  id: string,
+}
 
 /**
  * The GraphQL type equivalent of the People resource
  */
-const PersonType = new GraphQLObjectType({
+const PersonType: GraphQLObjectType<Person> = new GraphQLObjectType<Person>({
```

Comment:
@Urigo this doesn't make sense, I can't believe the type fidelity has fallen this much in `graphql`, migrating from flow to typescript it's even more apparent, I remember types in 14 at least more inference, am I doing something incorrectly here?

Reply:
cc @dotansimha @n1ru4l @saihaj